files/initramfs-tools/union-mount.j2

```diff
@@ -151,16 +165,21 @@ mount --bind ${rootmnt}/host/$image_dir/boot ${rootmnt}/boot
 
 ## Mount loop device or tmpfs for /var/log
 if $logs_inram; then
-  # NOTE: some platforms, when reaching initramfs stage, have a small
-  #       limit of mounting tmpfs partition, potentially due to amount
-  #       of RAM available in this stage. e.g. Arista 7050-qx32[s] and 7060-cx32s
-  set_tmpfs_log_partition_size
-  mount -t tmpfs -o rw,nosuid,nodev,size=${varlogsize}M tmpfs ${rootmnt}/var/log
-  [ -f ${rootmnt}/host/disk-img/var-log.ext4 ] && rm -rf ${rootmnt}/host/disk-img/var-log.ext4
+    # NOTE: some platforms, when reaching initramfs stage, have a small
+    #       limit of mounting tmpfs partition, potentially due to amount
+    #       of RAM available in this stage. e.g. Arista 7050-qx32[s] and 7060-cx32s
+    set_tmpfs_log_partition_size
+    mount -t tmpfs -o rw,nosuid,nodev,size=${varlogsize}M tmpfs ${rootmnt}/var/log
+    if [ -f ${rootmnt}/host/disk-img/var-log.ext4 ]; then
+        rm -rf ${rootmnt}/host/disk-img/var-log.ext4
+    fi
 else
-  [ -f ${rootmnt}/host/disk-img/var-log.ext4 ] && fsck.ext4 -v -p ${rootmnt}/host/disk-img/var-log.ext4 2>&1 \
-                                                  | gzip -c >> /tmp/fsck.log.gz
-  [ -f ${rootmnt}/host/disk-img/var-log.ext4 ] && mount -t ext4 -o loop,rw ${rootmnt}/host/disk-img/var-log.ext4 ${rootmnt}/var/log
+    if [ -f ${rootmnt}/host/disk-img/var-log.ext4 ]; then
+        fsck.ext4 -v -p ${rootmnt}/host/disk-img/var-log.ext4 2>&1 | gzip -c >> /tmp/fsck.log.gz
```

Comment:
Couldn't these 2 statements (`fsck` & `mount`) fit in a single `if` condition?
I don't believe `fsck.ext4` would remove the blob.
However if you prefer keeping it this way for now I'm fine with it.

Reply:
Good catch. Thanks!